Mage.Sets/src/mage/cards/b/BleedingEffect.java

```diff
@@ -0,0 +1,147 @@
+    private static final FilterControlledCreaturePermanent filterFirstStrike = new FilterControlledCreaturePermanent();
+    private static final FilterControlledCreaturePermanent filterDoubleStrike = new FilterControlledCreaturePermanent();
+    private static final FilterControlledCreaturePermanent filterDeathtouch = new FilterControlledCreaturePermanent();
+    private static final FilterControlledCreaturePermanent filterHexproof = new FilterControlledCreaturePermanent();
+    private static final FilterControlledCreaturePermanent filterIndestructible = new FilterControlledCreaturePermanent();
+    private static final FilterControlledCreaturePermanent filterLifelink = new FilterControlledCreaturePermanent();
+    private static final FilterControlledCreaturePermanent filterMenace = new FilterControlledCreaturePermanent();
+    private static final FilterControlledCreaturePermanent filterReach = new FilterControlledCreaturePermanent();
+    private static final FilterControlledCreaturePermanent filterTrample = new FilterControlledCreaturePermanent();
+    private static final FilterControlledCreaturePermanent filterVigilance = new FilterControlledCreaturePermanent();
+    
+    private static final FilterControlledCreaturePermanent filterCreatures = new FilterControlledCreaturePermanent();
+
+    static {
+        filterFirstStrike.add(new AbilityPredicate(FirstStrikeAbility.class));
+        filterFlying.add(new AbilityPredicate(FlyingAbility.class));
+        filterDeathtouch.add(new AbilityPredicate(DeathtouchAbility.class));
+        filterDoubleStrike.add(new AbilityPredicate(DoubleStrikeAbility.class));
+        filterHexproof.add(new AbilityPredicate(HexproofBaseAbility.class));
+        filterIndestructible.add(new AbilityPredicate(IndestructibleAbility.class));
+        filterLifelink.add(new AbilityPredicate(LifelinkAbility.class));
+        filterMenace.add(new AbilityPredicate(MenaceAbility.class));
+        filterReach.add(new AbilityPredicate(ReachAbility.class));
+        filterTrample.add(new AbilityPredicate(TrampleAbility.class));
+        filterVigilance.add(new AbilityPredicate(VigilanceAbility.class));
+    }
+
+    BleedingEffectEffect() {
+        super(Outcome.BoostCreature);
+        this.staticText = "creatures you control gain flying until end of turn if a creature card in your graveyard has flying." +
+                " The same is true for first strike, double strike, deathtouch, hexproof, indestructible, lifelink, menace, reach, trample, and vigilance.";
+    }
+
+    private BleedingEffectEffect(final BleedingEffectEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public BleedingEffectEffect copy() {
+        return new BleedingEffectEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+
+        // First strike
+        if (game.getBattlefield().containsControlled(filterFirstStrike, source, game, 1)) {
```

Comment:
um, isn't this "a creature card in your graveyard"?

also I really don't like the verbosity of duplicating the same code for each ability.

Try something like the following (please test it):

```java

        Map<Class<? extends Ability>, Ability> abilityMap = new HashMap<>();
        abilityMap.put(FirstStrikeAbility.class, FirstStrikeAbility.getInstance());
        abilityMap.put(FlyingAbility.class, FlyingAbility.getInstance());
        // the above should be made complete and declared static

        player.getGraveyard().stream()
                .map(game::getCard)
                .filter(Objects::nonNull)
                .flatMap(c -> c.getAbilities(game).stream())
                .map(Ability::getClass)
                .filter(abilityMap::containsKey)
                .forEach(c -> game.addEffect(new GainAbilityControlledEffect(
                        abilityMap.get(c),
                        Duration.EndOfTurn,
                        StaticFilters.FILTER_CONTROLLED_CREATURES), source));

```

Reply:
You're correct I'll update